### DISCUSSION TITLE
aria-controls pointing to an element that may be hidden

### DISCUSSION THREAD

**Discussion by o-alquimista** [w3c/aria#1601] [Q&A] [answered]
If you inspect the button named "Button with data-target" in the example at https://getbootstrap.com/docs/4.6/components/collapse/#example with the accessibility inspector built into Firefox, you'll notice that the collapsed area does not appear in the button's "Controller for" relationship list unless it is visible. Doesn't this mean that the accessibility effect given by `aria-controls` is totally ruined?

An example of such a situation:
```html
<button type="button" aria-controls="hiddenElement">show/hide</button>

<div id="hiddenElement" style="display: none">
  <span>toggle the visibility of this element</span>
</div>
```

I have Orca installed on Ubuntu, and that's the only screen reader I can test at the moment. It doesn't seem to be exposing this relationship to the user at all.

**Comment by JAWS-test** [comment, marked as answer]
> Orca: It doesn't seem to be exposing this relationship to the user at all.

This also applies to the screen readers JAWS and NVDA, which have decided not to output aria-controls

> Doesn't this mean that the accessibility effect given by aria-controls is totally ruined?

aria-controls should only be output if I can meaningfully navigate to the controlled element or area. If it is invisible, this is not the case, so I find Firefox's behavior fine

See: https://github.com/w3c/aria/issues/995
